lib/razor/cli/navigate.rb

```diff
@@ -5,14 +5,39 @@
 
 module Razor::CLI
   class Navigate
+    PRIORITY_RAZOR_API = "https://localhost:8151/api"
+    DEFAULT_RAZOR_API = "http://localhost:8150/api"
+    def self.default_api
```

Comment:
I think we could actually move the logic in set_api! (for the priority and default razor API cases) here and memoize the result.

This makes it more intuitive. set_api! would then read:

```
1. Set to the user-provided, if any
2. Set to the env. variable RAZOR_API
3. Set to the default
```

Reply:
You could also have the CLI output the default api to the user, and it would be consistent for PE installations and foss. As it is currently, the CLI will only output the FOSS endpoint as the default.

Reply:
As far as I can tell, optparse (which we use to display our help) doesn't allow lazy evaluation of strings. So that means before we start parsing what the user has input (`-u <url>` or not), we have to decide what the help string will be. So we'd either get an infinite loop, or a potentially superfluous call to our Priority URL.

Adding the accessor for `DEFAULT_RAZOR_API` here is an attempt to keep coupling from `Parse`=>`Navigate` loose.

Reply:
I don't understand how that prevents something like this:
```
def self.default_api
  if pe_razor_api_exists?
    return PE_RAZOR_API
  else
    return DEFAULT_RAZOR_API
end
```

^ And you can memoize the result in a class-variable if you want to avoid the (possible) case of checking pe_razor multiple times. This way, you can also avoid the whole verify_ssl thing that's currently done there. None of this depends on user input.

Reply:
If the user supplies `-u <URL>`, or has the `RAZOR_API` environment variable set, that HEAD request happens anyway. As the code stands now, that extra request is avoided.

Reply:
OK, the reason what I said won't work is chicken and egg -- we need an instance of `navigate` but we declare that instance after our `parse` class is created (since `create_resource` is an instance method, which I also don't think is good).

I think it would be useful to include the priority and default razor API in the help output (something like "default may be {priority} if that URL works, else it will be {default}"), because a PE user reading that help info. might find it confusing that their default RAZOR_API is the priority one despite the help text saying it is going to be the default one. Would it be bad to reference the constants directly? They are constants.

Also, I think having `default_api` is still useful, just we can make it an instance method and have it do something like
```
def default_api
  if priority_api_exists?
    return priority_api
  else
    return default_api
end
```

^ Something like that. That would make the set_api_url routine more readable (Step 3 would be a call to `default_api`). But I don't have a strong opinion on that -- I am fine with removing `default_api` too.

Reply:
I debated referencing both URLs; I see the appeal. I'll update the help message as well as the `default_api` method.